lib/hanami/events/adapter/memory_async.rb

```diff
@@ -11,7 +13,7 @@ def initialize(logger: nil, **)
         end
 
         def broadcast(event_name, payload)
-          @event_queue << { name: event_name, payload: payload }
+          @event_queue << { id: id_generator, name: event_name, payload: payload }
```

Comment:
Could be:
`@event_queue << { id: SecureRandom.uuid, name: event_name, payload: payload }`